src/components/modules/AuthHeader/AuthHeader.tsx

```diff
@@ -0,0 +1,41 @@
+import { useRouter } from "next/router";
+import React, { FC, ReactNode } from "react";
+
+import { Icon } from "@elements/Icon";
+
+import styles from "./AuthHeader.module.scss";
+
+interface AuthHeaderProps {
+    logoComponent: ReactNode;
+}
+
+export const AuthHeader: FC<AuthHeaderProps> = ({ logoComponent }) => {
+    const router = useRouter();
+
+    function handleBack() {
+        router.back();
+    }
+
+    function handleKeyDown(e: React.KeyboardEvent<HTMLDivElement>) {
+        if (e.key === "Enter") {
+            handleBack();
+        }
+    }
+    return (
+        <div className={styles.container}>
+            <div
+                tabIndex={0}
+                className={styles.backBtnContainer}
+                onClick={handleBack}
+                onKeyDown={handleKeyDown}
+                role="button"
+            >
+                <Icon name="arrowLeft" width={10} height={10} className={styles.arrowLeftIcon} />
+                <button type="button" className={styles.backBtnText}>
+                    back
+                </button>
+            </div>
```

Comment:
div with `role='button'` is kinda bad for semantics 
something like that would be better
```suggestion
            <button
                tabIndex={0}
                className={styles.backBtnContainer}
                onClick={handleBack}
                onKeyDown={handleKeyDown}
            >
                <Icon name="arrowLeft" width={10} height={10} className={styles.arrowLeftIcon} />
                <span className={styles.backBtnText}>back</span>
            </button>
```

Reply:
In this case, div acts like a button, and I left a button tag to fit his in one style. And to avoid errors I follow these rules: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/f0d2ddb65f21278ad29be43fb167a1092287b4b1/docs/rules/no-static-element-interactions.md